pages/chaoss/visualizations/contribs_by_action.py

```diff
@@ -0,0 +1,229 @@
+                                        size="sm",
+                                    ),
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ],
+                            align="center",
+                        ),
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+
+# callback for graph info popover
+@callback(
+    Output(f"popover-{PAGE}-{VIZ_ID}", "is_open"),
+    [Input(f"popover-target-{PAGE}-{VIZ_ID}", "n_clicks")],
+    [State(f"popover-{PAGE}-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+# callback for contributors by action graph
+@callback(
+    Output(f"{PAGE}-{VIZ_ID}", "figure"),
+    Output(f"check-alert-{PAGE}-{VIZ_ID}", "is_open"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"date-interval-{PAGE}-{VIZ_ID}", "value"),
+        Input(f"action-dropdown-{PAGE}-{VIZ_ID}", "value"),
+    ],
+    background=True,
+)
+def contribs_by_action_graph(repolist, interval, action):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=ctq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=ctq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.warning(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.warning(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph, False
+
+    # checks if there is a contribution of a specfic action in repo set
+    if not df["Action"].str.contains(action).any():
+        return dash.no_update, True
+
+    # function for all data pre processing
+    df = process_data(df, interval, action)
+
+    fig = create_figure(df, interval, action)
+
+    logging.warning(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig, False
+
+
+def process_data(df: pd.DataFrame, interval, action):
+
+    # convert to datetime objects rather than strings
+    df["created_at"] = pd.to_datetime(df["created_at"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created_at", axis=0, ascending=True)
+
+    # drop all contributions that are not the selected action
+    df = df[df["Action"].str.contains(action)]
+
+    return df
+
+
+def create_figure(df: pd.DataFrame, interval, action):
+    # time values for graph
+    x_r, x_name, hover, period = get_graph_time_values(interval)
+
+    # create plotly express histogram
+    fig = px.histogram(df, x="created_at", color_discrete_sequence=[color_seq[3]])
+
+    # creates bins with interval size and customizes the hover value for the bars
+    fig.update_traces(
+        xbins_size=interval,
+        hovertemplate=hover + "<br>Contributors: %{y}<br><extra></extra>",
```

Comment:
I'd also like to see the hover include the type of contribution being considered.